Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/PolDiffILLReduction.py

```diff
@@ -0,0 +1,876 @@
+                        except ValueError:
+                            CloneWorkspace(InputWorkspace=mtd[numor + '_' + direction], OutputWorkspace=name)
+                    else:
+                        list_pol.append('{0}_{1}'.format(numor, direction))
+                if average_detectors:
+                    Divide(LHSWorkspace=name, RHSWorkspace=CreateSingleValuedWorkspace(len(numors)), OutputWorkspace=name)
+                else:
+                    SumOverlappingTubes(','.join(list_pol), OutputWorkspace=name,
+                                        OutputType='1D', ScatteringAngleBinning=0.5, Normalise=True, HeightAxis='-0.1,0.1')
+                names_list.append(name)
+            DeleteWorkspaces(ws)
+            GroupWorkspaces(InputWorkspaces=names_list, OutputWorkspace=ws)
+        return ws
+
+    def _normalise(self, ws):
+        """Normalises the provided WorkspaceGroup to the monitor 1."""
+        monID = 100000 # monitor 1
+        monitor_indices = "{},{}".format(mtd[ws].getItem(0).getNumberHistograms()-2,
+                                         mtd[ws].getItem(0).getNumberHistograms()-1)
+        for entry_no, entry in enumerate(mtd[ws]):
+            mon = ws + '_mon'
+            ExtractSpectra(InputWorkspace=entry, DetectorList=monID, OutputWorkspace=mon)
+            if 0 in mtd[mon].readY(0):
+                raise RuntimeError('Cannot normalise to monitor; monitor has 0 counts.')
+            else:
+                if self._mode == 'TOF':
+                    Integration(InputWorkspace=mon, OutputWorkspace=mon)
+                Divide(LHSWorkspace=entry, RHSWorkspace=mon, OutputWorkspace=entry)
+                RemoveSpectra(entry, WorkspaceIndices=monitor_indices, OutputWorkspace=entry)
+                DeleteWorkspace(mon)
+        return ws
+
+    def _calculate_transmission(self, ws, beam_ws):
+        """Calculates transmission based on the measurement of the current sample, empty beam, and cadmium."""
+        # extract Monitor2 values
+        if 0 in mtd[ws].getItem(0).readY(0):
+            raise RuntimeError('Cannot calculate transmission; monitor has 0 counts.')
+        if 0 in mtd[beam_ws].readY(0):
+            raise RuntimeError('Cannot calculate transmission; beam monitor has 0 counts.')
+        Divide(LHSWorkspace=ws, RHSWorkspace=beam_ws, OutputWorkspace=ws)
+        return ws
+
+    def _subtract_background(self, ws, container_ws, transmission_ws):
+        """Subtracts empty container and cadmium scaled by transmission."""
+        if self._mode != 'TOF':
+            absorber_ws = self.getPropertyValue('AbsorberInputWorkspace')
+            if absorber_ws == "":
+                raise RuntimeError("Absorber input workspace needs to be provided for non-TOF background subtraction.")
+        for entry_no, entry in enumerate(mtd[ws]):
+            container_entry = mtd[container_ws].getItem(entry_no).name()
+            mtd[container_entry].setYUnit('Counts/Counts')
```

Comment:
This unit does not make sense. Why do you even set it?

Reply:
Every time `Divide` algorithm is invoked, it appends '/Counts' to the current 'Counts' YUnit string. This can be changed by setting to an empty string instead, as the label bears little importance at this point of reduction but if these labels do not agree, many subsequent algorithms raise exceptions.